scripts/index_aster_lpdaac.py

```diff
@@ -0,0 +1,640 @@
+ It reads the hdf files to create the Product/VRT/Dataset
+definitions, and write the datasets directly into an ODC database.
+It doesn't write out intermediate YAML files.
+
+The ODC Index datasets points to to the corresponding VRT files in order to access
+raster measurement data. The VRT file in turn points to the original `.hdf` file
+through `absolute path names` (Relative path names are not working at the moment,
+and it is advised that the VRT files must be generated for the final resident
+location of `.hdf` files.
+
+Each VRT file specify consistent set of bands from ASTER as a single product.
+For example, `vnir` sensors correspond to `aster_l1t_vnir` product, `tir`
+sensors correspond to `aster_l1t_tir` product, and `swir` sensors correspond
+to `aster_l1t_swir` product. The corresponding definitions of these product
+names and corresponding bands (with band names as identified in the original
+`hdf` file) are defined in the constant `PRODUCTS` in this script.
+
+It attempts to create stable UUIDs for the generated Datasets,
+based on the file path and modification time of the underlying HDF file Data
+as well as product name. Use following commands to create a product definition
+and add it to datacube, create a corresponding VRT file, and create a
+dataset definition and add it to datacube.
+
+
+::
+
+    ./index_nci_aster_lpdaac.py create-product
+                        --product aster_l1t_vnir /g/data/v10/ASTER_AU/2018.01.01
+    ./index_nci_aster_lpdaac.py create-vrt
+                        --product aster_l1t_vnir /g/data/v10/ASTER_AU/2018.01.01
+    ./index_nci_aster_lpdaac.py index-data
+                        --product aster_l1t_vnir /g/data/v10/ASTER_AU/2018.01.01
+
+
+::
+
+    psql -h agdcdev-db.nci.org.au
+    CREATE DATABASE aster_lpdaac WITH OWNER agdc_admin;
+    GRANT TEMPORARY, CONNECT ON DATABASE aster_lpdaac to public;
+
+aster_lpdaac.conf::
+
+    [datacube]
+    db_hostname: agdcdev-db.nci.org.au
+    db_port: 6432
+    db_database: aster_lpdaac
+
+::
+
+    datacube --config aster_lpdaac.conf system init
+
+::
+
+    for i in /g/data/v10/ASTER_AU/*; do
+        ./index_nci_aster_lpdaac.py --config aster_lpdacc.conf index-data
+                                    --product aster_l1t_vnir $i
+    done
+
+"""
+import json
+import logging
+import uuid
+from datetime import datetime, timezone
+from pathlib import Path
+import functools
+import yaml
+
+import click
+import numpy as np
+from osgeo import gdal, osr
+import rasterio
+
+from datacube import Datacube
+from datacube.index.hl import Doc2Dataset
+
+
+LOG = logging.getLogger(__name__)
+
+PRODUCTS = {'aster_l1t_vnir': {'ImageData2', 'ImageData1', 'ImageData3N'},
+            'aster_l1t_swir': {'ImageData4', 'ImageData5', 'ImageData6', 'ImageData7', 'ImageData8', 'ImageData9'},
+            'aster_l1t_tir': {'ImageData10', 'ImageData11', 'ImageData12', 'ImageData13', 'ImageData14'}}
+
+EXTRA_METADATA_PREFIXES = {
```

Comment:
It probably makes more sense to combine these two static dictionaries, they have the same keys anyway...

Reply:
OK done!